src/components/connect.js

```diff
@@ -3,6 +3,7 @@ import storeShape from '../utils/storeShape'
 import shallowEqual from '../utils/shallowEqual'
 import wrapActionCreators from '../utils/wrapActionCreators'
 import warning from '../utils/warning'
+import isFunction from 'lodash/isFunction'
```

Comment:
:+1: sounds good. The main reason I used `isFunction` was to make the code read similar to pattern matching. Like:
```haskell
case value of
  Function -> value
  PlainObject -> wrapActionCreators(value)
  _ -> defaultMapDispatchToProps
```
If that's a style we're interested in then maybe we can talk about it more, but for this PR i'll change it to use the `typeof` expression.